src/main/webapp/js/pages/index.js

```diff
@@ -37,18 +37,42 @@ function addScheduleToCalendar(schedule, courseInfo, selected) {
   });
 }
 
-document.querySelector('.course-list').addEventListener('click', () => {
-  const selected = CourseSelector.getSelected();
-  const courseInfo = CourseSelector.getCourseInfo();
-  // TODO(naaoli): Connect to algorithm servlet.
-  // Hard code return from algorithm servlet.
-  const sections1 = {};
-  selected.forEach(
-      course => {sections1[course] = courseInfo[course].sections[0]});
-  const sections2 = {};
-  selected.forEach(
-      course => {sections2[course] = courseInfo[course].sections[1]});
-  const schedules = [sections1, sections2];
+/**
+ * Returns a list of schedules with the selected courses
+ * @param {Array.<string>} selected JSON Array of the selected classes to
+ *     schedule
+ */
+async function getSchedules(selected) {
+  let response;
+  let scheduleObject;
+  const data = {selectedClasses: selected};
+  try {
+    response = await fetch('/api/scheduler', {
+      method: 'POST',
+      body: JSON.stringify(data),
+    });
+    scheduleObject = await response.json();
+    if (response.ok) {
+      return scheduleObject.schedules;
+    } else {
+      return null;
+    }
+  } catch (err) {
+    CourseSelector.createAlert('An error occurred', 'danger', courseContainer);
+    return null;
+  }
+}
+
+document.querySelector('.course-list').addEventListener('click', async () => {
+  const selected = CollegePlanner.getSelected();
+  const courseInfo = CollegePlanner.getCourseInfo();
```

Comment:
These also have to be changed to CourseSelector -- whenever you merge with master, make sure you test things on the devserver to make sure all of the modules still work.